internal/api/v1/organizations.go

```diff
@@ -161,3 +155,53 @@ func (oc OrganizationsController) Delete(w http.ResponseWriter, r *http.Request)
 
 	return nil
 }
+
+func deleteApps(cluster *kubernetes.Cluster, gitea *gitea.Client, org string) error {
+	apps, err := application.List(cluster, org)
+	if err != nil {
+		return err
+	}
+
+	const maxConcurrent = 100
+	errChan := make(chan error, maxConcurrent)
+	success := make(chan struct{})
+	stop := make(chan struct{})
+
+	go func() {
+		var wg sync.WaitGroup
+		// create a buffer, so there is no more than 'maxConcurrent' goroutines running at the same time
+		buffer := make(chan struct{}, maxConcurrent)
+	loop:
+		for _, app := range apps {
+			buffer <- struct{}{}
+
+			go func(app application.Application) {
+				defer wg.Done()
+				wg.Add(1)
+				err = application.Delete(cluster, gitea, org, app)
+				if err != nil {
+					errChan <- err
+				}
+				<-buffer
```

Comment:
This should better be defered too to ensure that no-matter what happens during the go routine call, the buffer will be freed in the end.

Reply:
done